system/Database/BaseResult.php

```diff
@@ -213,16 +213,16 @@ public function getResultArray(): array
 		// In the event that query caching is on, the result_id variable
 		// will not be a valid resource so we'll simply return an empty
 		// array.
-		if ( ! $this->resultID || $this->numRows === 0)
+		if (! $this->resultID || $this->numRows === 0)
 		{
 			return [];
 		}
 
-		if (($c = count($this->resultObject)) > 0)
+		if ($this->resultObject)
 		{
-			for ($i = 0; $i < $c; $i ++ )
+			foreach ($this->resultObject as $row)
 			{
-				$this->resultArray[$i] = (array) $this->resultObject[$i];
+				$this->resultArray[$i] = (array) $row;
```

Comment:
you forgot $i here

Reply:
i mean there should be no $i in this function

Reply:
dont know why the Travis build success but you forgot to remove the $i on the left hand array in both foreach loops ( l:225, l:268 )

Reply:
Ooops, fixed now. Thank you @puschie286